src/transport/types.d.ts

```diff
@@ -62,7 +63,7 @@ export type MultiaddrConnectionTimeline = {
 
 export type MultiaddrConnection = {
   sink: Sink;
-  source: () => AsyncIterable<Uint8Array>;
+  source: AsyncIterable<Uint8Array | BufferList>;
```

Comment:
What are the use-cases for returning a plain `BufferList`, are we doing this as an optimisation to avoid buffer concatenation? Do our code paths actually take advantage of this or do we end up concatenating anyway?

Reply:
We use `BufferList` because some modules we use (example: https://github.com/bustle/streaming-iterables) also support them and it is valid from users stand point, when they create libp2p protocols. Anyway, we don't use this internally anyway for optimisation and we end up concatenating as you mention. 
This is something that we can look into